### DISCUSSION TITLE
Active file is not in c++ source file

### DISCUSSION THREAD

**Discussion by Grutz09** [microsoft/vscode-cpptools#11504] [General]
This problem shows when I try to run it from VSCode.

So guys how can I fix this problem? like is there a problem with my installation process of msys2? or I just need to do something to fix this active file??

**Comment by bobbrow** [comment]
Active file is the one that has focus in VS Code.  (main.cpp in the picture below)

![image](https://github.com/microsoft/vscode-cpptools/assets/12818240/4ff6bcae-f455-4f67-9819-7d062c139c00)

For example, if you opened your launch.json file to make some changes and pressed F5 while that file was the one focused, you would get this error.  To fix it, click on the main.cpp tab (for example) and then press F5.

**Reply by Grutz09** [reply]
Still not effective sir. I'm clicking F5 while I'm in my cpp file but it's still the same

**Reply by bobbrow** [reply]
Can you share the exact steps you're taking to get to this error?  Are you using the "play" button in the editor, or do you have `tasks.json` and `launch.json`?

**Reply by Grutz09** [reply]
I do have tasks.json and launch.json but i don't know them so I use the play button in the editor the "run C/C++ file"

**Reply by bobbrow** [reply]
Do you get any output in the Terminal window, or is the error showing up in a notification message?  I'm fishing for more information because I don't know enough about what you're seeing to be able to be helpful.

For example, these are my steps:

1. I focus the file I want to run
2. I click the play button in the editor
3. I select the compiler I want to use to compile it
![image](https://github.com/microsoft/vscode-cpptools/assets/12818240/b9f38025-f626-4920-b319-65baade438e6)
4. The file builds. Here's the terminal output:
```
*  Executing task: C/C++: g++.exe build active file 

Starting build...
C:\msys64\mingw64\bin\g++.exe -fdiagnostics-color=always -g D:\test\simple\7.cpp -o D:\test\simple\7.exe

Build finished successfully.
 *  Terminal will be reused by tasks, press any key to close it. 
```
5. Then the built binary runs

**Reply by Grutz09** [reply]
Hello sir this is what  happens when I run it
![image](https://github.com/microsoft/vscode-cpptools/assets/140926634/a6249488-0ce6-4cfa-8b3b-fe8977f9c9c8)

**Reply by Grutz09** [reply]
![image](https://github.com/microsoft/vscode-cpptools/assets/140926634/2e8e5515-c35f-4490-9ede-5ac2519b93f8)
Good evening Sir! (it's evening here in my place). I just wanted to let you know that I followed your steps and it worked. Thank you so much sir! You're my life savior I'm really thankful.

**Reply by bobbrow** [reply]
I'm glad to hear you got it working.  From your screenshots, I'm wondering if the "Output" window was focused and that was causing the problem.
